node/cmd/guardiand/adminclient.go

```diff
@@ -236,13 +236,15 @@ func getPublicRPCServiceClient(ctx context.Context, addr string) (*grpc.ClientCo
 }
 
 func runSignWormchainValidatorAddress(cmd *cobra.Command, args []string) error {
+	ctx := context.Background()
+
 	guardianSignerUri := args[0]
 	wormchainAddress := args[1]
 	if !strings.HasPrefix(wormchainAddress, "wormhole") || strings.HasPrefix(wormchainAddress, "wormholeval") {
 		return errors.New("must provide a bech32 address that has 'wormhole' prefix")
 	}
 
-	guardianSigner, err := guardiansigner.NewGuardianSignerFromUri(guardianSignerUri, *unsafeDevnetMode)
+	guardianSigner, err := guardiansigner.NewGuardianSignerFromUri(ctx, guardianSignerUri, *unsafeDevnetMode)
```

Comment:
Context shouldn't be passed to constructors but to calls, such as `Sign` and `Verify` by the callers.
- https://go.dev/blog/context-and-structs#prefer-contexts-passed-as-arguments
- https://go.dev/blog/context-and-structs#storing-context-in-structs-leads-to-confusion

That said, I think we can keep this as is and simply add comments explaining why this is a special case, i.e. you need a context for calling AWS KMS during signer initialization.

Reply:
Alternatively, the initialization may not ask for a parent Context and simply manage its own, e.g. a `Context.WithTimeout(timeout)` should suffice.